### ISSUE TITLE
Support for user defined Schema

### ISSUE THREAD

**Issue by ianagius** [polyjdbc/polyjdbc#20]
At the moment polyjdbc use the db current default schema to create tables. This needs to be fixed to allow users to select which schema polyjdbc should use.

**Comment by adamdubiel** [comment]
Can you be bit more specific? Show some code sample where the change should be applied?

**Comment by ianagius** [comment]
We are currently using JaVers and we need a modification to PolyJDBC to allow us to create tables in a user defined schema. We were thinking of something on the following lines:

Modify builder to register a user defined schema (should be optional ofcourse):

public static PolyJDBCBuilder polyJDBC(Dialect dialect, String schemaName) {
        return new PolyJDBCBuilder(dialect, schemaName);
    }

and then modify polyjdbc to handle the schema when creating/altering/deleting tables.

**Comment by adamdubiel** [comment]
Okay, that sounds reasonable, i bet @bartoszwalacik will help us :) I will try to work on this in following week.

**Comment by bartoszwalacik** [comment]
I bet that a change in polyjdbc code would be rather small

**Comment by adamdubiel** [comment]
So i wanted to write some code, but before i do this - what is wrong with setting schema in database url? For H2:

```
jdbc:h2:mem:test;schema=my_schema
```

In postgres:

```
jdbc:postgresql://localhost:5432/polly?searchpath=my_schema
```

Do you need PolyJDBC to support switching between schemas on the fly?

**Comment by bartoszwalacik** [comment]
Switching on the fly is not required. We need to test is it possible to log on user X giving schema name Y and have tables created in schema Y ( without prefixing the table names in ddl generated by Polly)

**Comment by ianagius** [comment]
We are preparing a pull request for this which should be available this week.  

While preparing for it we noticed that one of the constructors of SchemaInspectorImp was not invoking extractMetadata(..) (see snippet below).

```
    SchemaInspectorImpl(Transaction transaction) {
        this(transaction, Locale.ENGLISH);
        extractMetadata(transaction);
    }

    SchemaInspectorImpl(Transaction transaction, Locale locale) {
        this.transaction = transaction;
        this.locale = locale;
    }
```

Is this on purpose ?  Shouldn't extractMetadata(..) be invoked on each instantiation of SchemaInspectorImpl so that metadata is not null ?

**Comment by ianagius** [comment]
pull request 20 done

**Comment by bartoszwalacik** [comment]
thnx, we will see over it

btw. i'm not sure about extractMetadata(), maybe Adam will help

**Comment by ianagius** [comment]
In the pull request, extractMetadata(..) was added to the missing constructor. Without it, if a call uses that constructor,  variables connection, metadata and catalog remain null and the others methods will automatically fail (null pointer exception).

**Comment by adamdubiel** [comment]
I looked at the PR, looks good to me, except for some wierd formatting in few places:

```
public method
{
   ...
}
```

Would be good if you kept consistent with existing code base. I will continue with review, should be finished today.

**Comment by ianagius** [comment]
apologies for that. Fixed formatting.

**Comment by ianagius** [comment]
By any chance, did you have any time to look at the PR?

**Comment by bartoszwalacik** [comment]
I did the CR, please follow it

**Comment by ianagius** [comment]
Fixed reported issues

**Comment by adamdubiel** [comment]
Released in 0.6.3 version - thanks for the PR :)
